modyn/trainer_server/internal/trainer/remote_downsamplers/abstract_matrix_downsampling_strategy.py

```diff
@@ -71,6 +71,9 @@ def inform_samples(
     ) -> None:
         batch_size = len(sample_ids)
         assert self.matrix_content is not None
+        if forward_output.dim() == 1:
+            forward_output = forward_output.unsqueeze(1)
+            target = target.unsqueeze(1)
```

Comment:
This code is basically everywhere. Can we have a utils function `_unsequeeze_if_required` or something, and in that utils function have a brief comment that this is necessary for the cases where we have binary classification with sigmoid instead of softmax?

Reply:
Also, is this sufficient? I did not think this through, but all of these downsamplers were written for the softmax-case, i.e., 1 entry in a probabilty vector per class. I wonder whether this also works here, or if we need to "simulate" a softmax like you did in uncertainty implementation: 
```
# sigmoid
 preds = torch.cat((1 - preds, preds), dim=1)
```

this also applies to the other downmsaplers (craig, ...)

Reply:
I checked all downsampler implementations only uncertainty based downsamplers assume that the output is in the form of "1 entry in a probabilty vector per class". For others they are based on gradients or embedding which does not depend on the output form.

For the util function: sure I will make the change